keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ComparableLinksTests.kt

```diff
@@ -287,11 +287,11 @@ class ComparableLinksTests : JUnit5Minutests {
         before {
           every { repository.artifactVersions(releaseArtifact, limit) } returns versions.toArtifactVersions(
             releaseArtifact,
-            "stash"
+            "https://stash"
```

Comment:
Someone please review if it was critical that this work with commit urls that are protocol less. I think that was simply an oversight in writing the test.

Reply:
I don't know the code here to say one way or another, but putting this change in a separate commit would help clarify this....and make it easier to see that tests continue to pass before the other changes in the pr.

Reply:
Good point. I will restructure this.

Reply:
Now in explicit commit. Still calling attention to this though here so that it gets eyes :eyes:

Reply:
previously, we didnt parse the link to extract the type so https in the source type didnt matter. better version for this change to go is probably here, where the link to the git artifact gets created: https://github.com/spinnaker/keel/blob/cf94218db7fc757c260dd5d866d478857dae0292/keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ComparableLinksTests.kt#L157

Reply:
Are you suggesting that we update the mock generation code to convert "stash" -> "https://stash" rather than just use the string literal  "https://stash" below? What is the real world situation? Are these artifacts going to have protocol less sources? I like modifying the literal because the test reads more explicitly.

Reply:
yeah either that or have `toArtifactVersions` and the `git` mock receive another optional parameter for `link`. Problem is, the second parameter for `toArtifactVersion` is the source type, and passing the protocol scheme as part of the source type makes it look weird.

Reply:
I understand what you mean and I agree, however aren't the original github tests passing protocol in via the second parameter? https://github.com/spinnaker/keel/blob/cf94218db7fc757c260dd5d866d478857dae0292/keel-core/src/test/kotlin/com/netflix/spinnaker/keel/services/ComparableLinksTests.kt#L620 should we clean all of these up for consistency in a separate PR?

Reply:
yep, I agree with the inconsistencies, and it is a good idea to fix it while we are at it.